gitpack/main/lib/claude.py

```diff
@@ -0,0 +1,172 @@
+from anthropic import Anthropic
+import json, re
+from django.conf import settings
+import logging
+
+class ClaudeHelper:
+    def __init__(self):
+        self.client = Anthropic(
+            api_key=settings.ANTHROPIC_API_KEY
+        )
+
+    def get_code_review(self, files_changed):
+        all_code = ""
+        files_changed_dict = {}
+
+        # Loop through each changed file
+        for file in files_changed:
+            if any(re.match(pattern, file.filename) for pattern in settings.CODE_REVIEW_IGNORE_PATTERNS):
+                continue
+            file_patch = file.patch
+            files_changed_dict[file.filename] = file_patch
+            all_code += f"File path: {file.filename}\n\nContent:\n{file.patch}\n\n"
+
+        logging.debug('files_changed_dict:\n\n%s\n\n', json.dumps(files_changed_dict, indent=4, sort_keys=True))
+
+        prompt = f"""Review the following code and provide feedback with positive aspects and areas for improvement.
+
+You must respond with a RFC8259 compliant JSON object following this exact format. Do not include any other text or explanations:
+
+{{
+    "inline_feedback": [
+        {{
+            "file_path": "relative path of the file",
+            "code_lines": ["array of actual code lines receiving feedback"],
+            "start_side": "LEFT or RIGHT",
+            "start_line": "absolute line number",
+            "end_side": "LEFT or RIGHT", 
+            "end_line": "absolute line number",
+            "feedback": "specific feedback for this code section",
+            "suggested_code_changes": "optional suggested code improvements"
+        }}
+    ],
+    "summary": {{
+        "summary": "overall code review summary",
+        "positives": ["array of positive aspects"],
+        "improvements": ["array of areas needing improvement"]
+    }}
+}}
+
+Only provide feedback for significant and impactful issues. Avoid trivial feedback.
+For multi-line code sections, provide feedback in a single feedback object.
+
+=== Code to review ===
+
+{all_code}
+
+=== End code ===
+
+Remember to provide only the JSON response with no additional text."""
+
+        response = self.client.messages.create(
+            model="claude-3-opus-20240229",
+            max_tokens=1500,
+            messages=[{
+                "role": "user",
+                "content": prompt
+            }],
+            system="You are a Staff or Principal level Software Engineer performing code reviews. Focus only on high-quality, impactful feedback."
+        )
+        
+        response_content = response.content[0].text
+        try:
+            response_json = json.loads(response_content)
```

Comment:
Our current implementation is better because it:

Has a fallback mechanism for JSON in code blocks
Provides detailed error logging
Raises appropriate exceptions
Maintains the contract that get_code_review must return valid feedback